xmtp/src/storage/encrypted_store/mod.rs

```diff
@@ -16,54 +16,44 @@ pub mod schema;
 
 use log::warn;
 use rand::RngCore;
-use std::{ops::DerefMut, sync::Mutex};
-
-use self::schema::accounts;
-use self::{models::*, schema::messages};
-use crate::account::Account;
-use crate::{Errorer, Fetch, KeyStore, StorageError, Store};
-use diesel::{connection::SimpleConnection, prelude::*, Connection};
-use thiserror::Error;
+
+use self::{
+    models::*,
+    schema::{accounts, messages},
+};
+use crate::{account::Account, Errorer, Fetch, KeyStore, Store};
+use diesel::{
+    connection::SimpleConnection,
+    prelude::*,
+    r2d2::{ConnectionManager, Pool, PooledConnection},
+};
+
 use xmtp_cryptography::utils as crypto_utils;
 
 use diesel_migrations::{embed_migrations, EmbeddedMigrations, MigrationHarness};
-pub const MIGRATIONS: EmbeddedMigrations = embed_migrations!("./migrations/");
 
-#[derive(Debug, Error, PartialEq)]
-pub enum EncryptedMessageStoreError {
-    #[error("Diesel connection error")]
-    DieselConnectError(#[from] diesel::ConnectionError),
-    #[error("Diesel result error")]
-    DieselResultError(#[from] diesel::result::Error),
-    #[error("Either incorrect encryptionkey or file is not a db")]
-    DbInitError(String),
-    #[error("could not set encryption key")]
-    EncryptionKey,
-    #[error("could not generate encryptionKey")]
-    EncryptionKeyGen,
-    #[error("Mutex Poisioned: {0}")]
-    PoisionError(String),
-    #[error("unknown storage error")]
-    Unknown,
-}
+use super::StorageError;
+pub const MIGRATIONS: EmbeddedMigrations = embed_migrations!("./migrations/");
 
 pub type EncryptionKey = [u8; 32];
 
-#[derive(Default)]
+#[derive(Default, Clone, Debug)]
 pub enum StorageOption {
     #[default]
     Ephemeral,
     Persistent(String),
 }
 
 #[allow(dead_code)]
+#[derive(Clone)]
 /// Manages a Sqlite db for persisting messages and other objects.
 pub struct EncryptedMessageStore {
```

Comment:
I'm tempted to rename this to `SqliteStorage` or something a little more generic. Wdyt?

Reply:
I think that is prudent. EncryptedMessageStore doesn't quite fit the direction this struct has been heading.

Reply:
I'll do that in a subsequent PR. The churn on this one is big enough already